lib/bounces.js

```diff
@@ -39,22 +57,34 @@ module.exports = function (log) {
       else if (message.complaint && message.complaint.complaintFeedbackType === 'abuse') {
         recipients = message.complaint.complainedRecipients
       }
-      for (var i = 0; i < recipients.length; i++) {
-        var email = recipients[i].emailAddress
-        log.info({ op: 'handleBounce', email: email, bounce: !!message.bounce })
-        log.increment('account.email_bounced')
-        db.emailRecord(email)
-          .done(
-            deleteAccountIfUnverified,
-            gotError.bind(null, email)
-          )
-      }
-      message.del()
+      var p = P.resolve()
+      recipients.forEach(function(recipient) {
+        var email = recipient.emailAddress
+        p = p.then(
```

Comment:
This looks a little strange to me, wouldn't the function return when the last promise is resolved, or is that intended? Would using [Promise.reduce](http://bluebirdjs.com/docs/api/promise.reduce.html) simplify this section?